### PR TITLE
Support 1.21.4

### PR DESCRIPTION
**Pull request by Quozul**
This pull requests is based on the work from @BoomEaro in their pull request #98 

The PR #98 should be merged before this one.

As the protocol wiki is not updated yet to 1.21.4, I may be missing some things.

### CODE REVIEW

**Comment by BoomEaro** [comment]
Where did you find a fully prepared snbt codec file for 1.21.4? Because of this I had difficulties updating to 1.21.4 and I was going to send completely different registers to the client, as the vanilla server does

**Comment by Quozul** [comment]
I did not find one, just hacked something in an hour until it worked. This may be totally incomplete.

File: src/main/java/ua/nanit/limbo/protocol/packets/configuration/PacketPlayerLoaded.java
Comment (BoomEaro) [inline src/main/java/ua/nanit/limbo/protocol/packets/configuration/PacketPlayerLoaded.java:1]: This packet is called ClientSettings and in this particular case there seems to be no point in decoding it

**Review by BoomEaro — CHANGES_REQUESTED** [review]

File: src/main/java/ua/nanit/limbo/protocol/packets/configuration/PacketPlayerLoaded.java
Comment (Quozul) [reply src/main/java/ua/nanit/limbo/protocol/packets/configuration/PacketPlayerLoaded.java:1]: Where did you find the name of ClientSettings? Is there documentation available on this new packet?

**Comment by Potothingi** [comment]
I'm looking forward to this becoming stable.

You can refer to the other Limbo github:
https://github.com/LOOHP/Limbo/commit/61331a3ea16f6bd732176723f3c9cc5b58ceb84b

**Comment by Quozul** [comment]
It is going to take some time to rewrite all of this manually from the other Limbo...

File: src/main/java/ua/nanit/limbo/protocol/packets/configuration/PacketPlayerLoaded.java
Comment (BoomEaro) [reply src/main/java/ua/nanit/limbo/protocol/packets/configuration/PacketPlayerLoaded.java:1]: Not long ago, the popular wiki for minecraft protocol - wiki.vg was closed, but the information is transferred to a new wiki here:
https://minecraft.wiki/w/Minecraft_Wiki:Projects/wiki.vg_merge/Protocol

The packet with the number 0x00 in the configuration phase of Serverbound is a ClientSettings packet, in addition to the fact that it has a locale field, your packet is missing 8 more fields.
And as I already said, decoding this packet is not required at all

**Comment by BoomEaro** [comment]
If you don't mind, I can do the upgrade to 1.21.4 myself in my branch using a different approach and using different registers (without using snbt for the most part)

At this point all that is required to update to 1.21.4 is to update the mappings to 1.21.4 and the snbt codec. I am going to ditch the current codecs in favor of what the truly vanilla server sends to the client.

**Comment by Quozul** [comment]
Sure, go ahead! I will close this PR once you're done.

**Pull request closed without merging**